### DISCUSSION TITLE
Procedure parameters changing

### DISCUSSION THREAD

**Discussion by AndreiTapai** [littleK0i/SnowDDL#100] [Q&A]
Curious about the proper way to handle signatures changing for procedures. We recently had an issue where we removed an optional parameter from a procedure. This then failed when running snowddl against our database because apparently optional parameters do not count when resolving the function signature. I was perhaps naively expecting a CREATE OR REPLACE to work here. What is the proper handling of situations like this? Do we need to provide a script that will explicitly drop the procedure before snowddl runs?

**Comment by littleK0i** [comment]
Since `PROCEDURE` uses [short-hash](https://docs.snowddl.com/guides/other/short-hash) approach to change management, any change should trigger `CREATE OR REPLACE`.

Signature validation does not have any specific conditions around "optional" arguments as well: https://github.com/littleK0i/SnowDDL/blob/master/snowddl/parser/abc_parser.py#L62-L90

There is a chance Snowflake secretly changed something in one of latest updates. Could you share a specific example of config change which does not trigger `CREATE OR REPLACE` or `DROP`, and I'll take a look.

Thank you.

**Comment by littleK0i** [comment]
As an initial check, did you update both `arguments` in config and name of YAML file itself?

SnowDDL should still detect and complaint in case of mismatch, but maybe there is an edge case which is not covered.
